lib/phoenix_live_view.ex

```diff
@@ -473,13 +473,26 @@ defmodule Phoenix.LiveView do
       require Phoenix.LiveView.Renderer
       @before_compile Phoenix.LiveView.Renderer
 
+      @phoenix_live_opts opts
+      Module.register_attribute(__MODULE__, :mount, accumulate: true)
+      @before_compile Phoenix.LiveView
+    end
+  end
+
+  defmacro __before_compile__(env) do
+    opts = Module.get_attribute(env.module, :phoenix_live_opts, [])
+    at_mount = Module.get_attribute(env.module, :mount, [])
+
+    quote bind_quoted: [opts: opts, at_mount: at_mount] do
       @doc false
-      def __live__, do: unquote(Macro.escape(Phoenix.LiveView.__live__(__MODULE__, opts)))
+      def __live__ do
+        unquote(Macro.escape(Phoenix.LiveView.__live__(__MODULE__, at_mount, opts)))
```

Comment:
We can call just:

```suggestion
        unquote(Macro.escape(__live__(__MODULE__, at_mount, opts)))
```

And make `__live__` private (I don't know why it wasn't before). Perhaps even merge `__before_compile__` and `__live__`.

Reply:
Done, but I would definitely appreciate another look 👀